File: twoSum.js
var twoSum = function(nums, target) {
    const x = {}; // to store the indices of the numbers
    // Iterate through the array
    for (let i = 0; i < nums.length; i++) {
        // Calculate the complement of the current number that would sum up to the target
        const z = target - nums[i];
        if (z in x) {
            // If the complement exists in the map, return the indices of the current number and the complement
            return [x[z], i];
        } else {
            // Otherwise, store the index of the current number
            x[nums[i]] = i;
        }
    }
    return [];  // in case no solution found, just for safety
};

File: twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        x = {} # Dictionary to store the value and its index
        for i,v in enumerate(nums): # Enumerate gives us both index and value
            # Check if the complement (target - v) exists in the dictionary
            if(target - v in x): 
                # If it exists, return the indices of the two numbers
                # The first index is from the dictionary, the second is the current index
                return [x[target - v],i]
            else:
                # If it does not exist, store the current value and its index in the dictionary
                x[v] = i
        # If no solution is found, return an empty list
        return []